pkg/streamingpromql/planning/core/function_call.go

```diff
@@ -128,3 +129,15 @@ func (f *FunctionCall) OperatorFactory(children []types.Operator, timeRange type
 		return nil, compat.NewNotSupportedError(fmt.Sprintf("'%v' function", f.FunctionName))
 	}
 }
+
+func (f *FunctionCall) ResultType() (parser.ValueType, error) {
+	if _, ok := functions.InstantVectorFunctionOperatorFactories[f.FunctionName]; ok || f.FunctionName == "absent" || f.FunctionName == "absent_over_time" {
```

Comment:
Bit of a nit, but I dislike we have to keep this special case in multiple places.

We should add a comment at least.

But I wonder if we can make this better. Maybe some kind of dummy entry in `functions.InstantVectorFunctionOperatorFactories` ?

Reply:
I worry that a dummy entry in `functions.InstantVectorFunctionOperatorFactories` might cause other problems or confusion.

I'm not expecting to need further instances of this special case anywhere else, if that helps allay your concerns.

Reply:
I think if it panics when used incorrectly it could work, but I'm nit picking so I'm okay with it as is, but would love a comment on the line please.